src/status_im/extensions/ethereum.cljs

```diff
@@ -234,12 +234,39 @@
                         :on-failure on-failure}]
      (wrap-with-resolution db json-rpc-args :to execute-send-transaction))))
 
-(defn- parse-log [{:keys [address transactionHash blockHash transactionIndex topics blockNumber logIndex removed data]}]
-  (merge {:data    data
+(defn- event-topic-enc [event params]
+  (let [eventid (str event "(" (string/join "," params) ")")]
+    (abi-spec/sha3 eventid)))
+
+; Return a vector without elements of Y
+(defn diff-vectors [X Y]
+  (reduce #(remove (fn [x] (= x %2)) %1) X Y))
+
+; Return a map with all data params decoded
+(defn- decode-data [data events-hints first-topic indexed]
+  (let [index-event (.indexOf (mapv #(event-topic-enc (% :event) (% :params)) events-hints) first-topic)
```

Comment:
Use of `.indexOf` is suspicious as you are calling it on a clojure object, not JS.

Reply:
Changed for `string/index-of`, thanks!